pkg/remote/crd_upgrade.go

```diff
@@ -116,23 +117,30 @@ func fetchCrdsAndUpdateKymaAnnotations(ctx context.Context, controlPlaneClient C
 	return crdUpdated, nil
 }
 
-func fetchCrds(ctx context.Context, controlPlaneClient Client, runtimeClient Client, plural string) (
+func fetchCrds(ctx context.Context, controlPlaneClient Client, runtimeClient Client, plural string,
+	kcpCrdsCache *internal.CustomResourceDefinitionCache) (
 	*v1extensions.CustomResourceDefinition, *v1extensions.CustomResourceDefinition, error,
 ) {
-	crd := &v1extensions.CustomResourceDefinition{}
 	crdFromRuntime := &v1extensions.CustomResourceDefinition{}
-	err := controlPlaneClient.Get(
-		ctx, client.ObjectKey{
-			// this object name is derived from the plural and is the default kustomize value for crd namings, if the CRD
-			// name changes, this also has to be adjusted here. We can think of making this configurable later
-			Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group),
-		}, crd,
-	)
-	if err != nil {
-		return nil, nil, err
+
+	kcpNamespacedNameCrd := client.ObjectKey{
+		// this object name is derived from the plural and is the default kustomize value for crd namings, if the CRD
+		// name changes, this also has to be adjusted here. We can think of making this configurable later
+		Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group)}
```

Comment:
You should not use self made name for `client.ObjectKey`, ObjectKey should be only used for go client to fetch k8s resources with its name and namespace. That's another reason why you should not use it as cache key.